File: Python/ProjectEuler_09.py
# Special Pythagorean Triplet -- Solved
# Solution = 31875000

from Timer import Timer
Stopwatch = Timer()

Solution = 0

for i in range(1,1001):
    for j in range(1,1001):

        k = 1000 - i - j
        # Alt. Expression: k = (i**2 + j**2)**0.5

        if i**2 + j**2 == k**2 and i + j + k == 1000:
            Solution = i*j*k
            break;

print('The product of the pythagorean triplet whose sum is 1000 is: ', Solution)
Stopwatch.stop()

File: MATLAB/ProjectEuler_09.m
%% Special Pythagorean Triplet -- Solved
clear, close, clc, disp(mfilename), tic;

TripletCheck = @(a,b,c) (a^2+b^2==c^2 && a<b && a+b+c==1000);
SolutionTriplet = [];
Limit = 1000;

for i = 1:Limit
    for j = 1:Limit
        k = Limit-(i+j);
        if TripletCheck(i,j,k)
            SolutionTriplet = [i j k];
            break;            
        end
    end
end

fprintf('The product of the pythagorean triplet whose sum is 1000 is: %d\n', prod(SolutionTriplet)); toc;